### PR TITLE
pythonPackages.pyro-ppl: fix build, unmark as broken

### PR DESCRIPTION
**Pull request by georgewhewell**
###### Motivation for this change
pythonPackages.pyro-ppl is broken

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
@ofborg eval

File: pkgs/development/python-modules/pyro-ppl/default.nix [hunk 1]
```diff
@@ -34,6 +35,5 @@ buildPythonPackage rec {
     homepage = "http://pyro.ai";
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ teh ];
```
Comment (teh) [inline pkgs/development/python-modules/pyro-ppl/default.nix:37]: please add yourself as a maintainer :)

File: pkgs/development/python-modules/pyro-api/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ buildPythonPackage, fetchPypi, lib }:
```
Comment (teh) [inline pkgs/development/python-modules/pyro-api/default.nix:1]: I think adding this should go into a separate commit like "python3Packages.pyro-api: init at 0.1.1".

**Review by teh — COMMENTED** [review]
small fixes, otherwise LGTM. Thanks for picking this up!

File: pkgs/development/python-modules/pyro-api/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "pyro-api";
+
+  src = fetchPypi {
+    inherit version pname;
+    sha256 = "0rhd7p61pf2vvflbdixp7sygblvvl9qbqavxj27910lr79vl4fdz";
+  };
+
+  checkPhase = ''
+    python -c 'import pyroapi'
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyro-api/default.nix:14]:
```suggestion
  pythonImportsCheck = [ "pyroapi" ];
```

File: pkgs/development/python-modules/pyro-api/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  meta = {
+    description = "Generic API for dispatch to Pyro backends.";
+    homepage = "http://pyro.ai";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ georgewhewell ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyro-api/default.nix:20]: You are missing platforms.

File: pkgs/development/python-modules/pyro-api/default.nix (hunk 2 above)
Comment (georgewhewell) [reply pkgs/development/python-modules/pyro-api/default.nix:1]: done

File: pkgs/development/python-modules/pyro-api/default.nix (hunk 4 above)
Comment (georgewhewell) [reply pkgs/development/python-modules/pyro-api/default.nix:21]: added

File: pkgs/development/python-modules/pyro-api/default.nix (hunk 3 above)
Comment (georgewhewell) [reply pkgs/development/python-modules/pyro-api/default.nix:14]: i tried this but it still runs the original `checkPhase` which attempts to run tests, which fail because they depend on `pyro-ppl` (which depend on this package). Not sure how this can be resolved

Comment (jonringer) [reply pkgs/development/python-modules/pyro-api/default.nix:14]: you also want to do `doCheck = false;` and leave a comment as to why they are not enabled

File: pkgs/development/python-modules/pyro-ppl/default.nix [hunk 5]
```diff
@@ -1,15 +1,16 @@
 { buildPythonPackage, fetchPypi, lib, pytorch, contextlib2
-, graphviz, networkx, six, opt-einsum, tqdm }:
+, graphviz, networkx, six, opt-einsum, tqdm, pyro-api }:
```
Comment (jonringer) [inline pkgs/development/python-modules/pyro-ppl/default.nix:2]: NIT
```suggestion
, graphviz, networkx, six, opt-einsum, tqdm, pyro-api }:

```

File: pkgs/development/python-modules/pyro-ppl/default.nix [hunk 6]
```diff
@@ -33,7 +34,7 @@ buildPythonPackage rec {
     description = "A Python library for probabilistic modeling and inference";
     homepage = "http://pyro.ai";
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ teh ];
-    broken = true;
+    platforms = lib.platforms.unix;
```
Comment (jonringer) [inline pkgs/development/python-modules/pyro-ppl/default.nix:37]: `buildPythonPackage` will add this for you
```suggestion
```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/pyro-ppl/default.nix:37]: Is this only required if the package is linux or darwin only?

File: pkgs/development/python-modules/pyro-api/default.nix (hunk 3 above)
Comment (georgewhewell) [reply pkgs/development/python-modules/pyro-api/default.nix]: done

File: pkgs/development/python-modules/pyro-ppl/default.nix (hunk 6 above)
Comment (georgewhewell) [reply pkgs/development/python-modules/pyro-ppl/default.nix]: removed again

File: pkgs/development/python-modules/pyro-ppl/default.nix (hunk 5 above)
Comment (georgewhewell) [reply pkgs/development/python-modules/pyro-ppl/default.nix:2]: applied

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 104269 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>5 packages built:</summary>  <ul>    <li>python27Packages.pyro-api</li>    <li>python37Packages.pyro-api</li>    <li>python37Packages.pyro-ppl</li>    <li>python38Packages.pyro-api</li>    <li>python38Packages.pyro-ppl</li>  </ul></details>
